src/server/pkg/ppsutil/util.go

```diff
@@ -156,6 +156,10 @@ func GetPipelineInfo(pachClient *client.APIClient, ptr *pps.EtcdPipelineInfo) (*
 		return nil, fmt.Errorf("could not unmarshal PipelineInfo bytes from PFS: %v", err)
 	}
 	result.State = ptr.State
+	result.Stopped = ptr.Stopped
+	if ptr.Stopped {
+		result.State = pps.PipelineState_PIPELINE_PAUSED
+	}
```

Comment:
This seems like the easiest way to report the `PAUSED` state ... since I'm not sure how to enforce an invariant on the `etcPipelineInfo`'s state at write time.

Reply:
So I don't think it should be an invariant that `Stopped -> State == PAUSED`. I think it makes sense that there's a window between when you stop the pipeline and when the master actually picks it up during which Stopped is true but `State != PAUSED`